pytest_inmanta/plugin.py

```diff
@@ -1358,6 +1392,143 @@ def finalize_all_handlers(self) -> None:
+
+        ctx = handler.HandlerContext(resource, dry_run=dry_run)
+        h.execute(ctx, resource, dry_run)
+        self.finalize_context(ctx)
+        self.finalize_handler(h)
+
+        out = DeployResultV2(resource=resource, ctx=ctx, handler=h)
+        if expected_status is not None:
+            out.assert_status(expected_status)
+
+        return out
+
+    def dryrun_resource_v2(
+        self,
+        resource_type: str,
+        run_as_root: bool = False,
+        expected_status: Optional[const.ResourceState] = DEFAULT,
+        **filter_args: object,
+    ) -> "DeployResultV2":
+        return self.deploy_resource_v2(
+            resource_type, run_as_root, True, expected_status, **filter_args
+        )
+
+
+@dataclass
+class DeployResultV2:
+    ctx: HandlerContext
+    handler: ResourceHandler
+    resource: Resource
+
+    # status
+    def assert_status(
+        self,
+        status: const.ResourceState = const.ResourceState.deployed,
+        change: const.Change = None,
+    ):
+        ctx = self.ctx
+        if ctx.status != status:
+            print("Deploy did not result in correct status")
+            print("Requested changes: ", ctx._changes)
+            for log in ctx.logs:
+                print("Log: ", log._data["msg"])
+                print(
+                    "Kwargs: ",
+                    [
+                        "%s: %s" % (k, v)
+                        for k, v in log._data["kwargs"].items()
+                        if k != "traceback"
+                    ],
+                )
+                if "traceback" in log._data["kwargs"]:
+                    print("Traceback:\n", log._data["kwargs"]["traceback"])
+
+        assert ctx.status == status
+        if change is not None:
+            assert ctx._change == change
+        self.assert_consistent_status()
+
+    # Discovery
+    @property
+    def discovered_resources(self) -> list[object]:
+        return self.handler._client.discovered_resources
+
+    # Logs
+    @property
+    def logs(self) -> list[LogLine]:
+        return self.ctx.logs
+
+    def assert_has_logline(self, matches: str) -> LogLine:
+        """
+         Assert the handler logged a log line matching the pattern
+
+        :return: that logline
+        """
+        pattern = re.compile(matches)
+        for logline in self.logs:
+            if pattern.search(logline.msg):
+                return logline
+        assert False, f"No line found matching {pattern}"
+
+    # changes
+    @property
+    def changes(self) -> Dict[str, AttributeStateChange]:
+        return self.ctx.changes
+
+    def assert_no_changes(self) -> None:
+        """Assert that the diff produced no changes"""
+        assert not self.changes
+
+    def assert_consistent_status(self):
+        """Make sure we report change and doing changes consistently"""
+        if self.ctx.status != const.ResourceState.deployed:
+            return
+
+        if bool(self.ctx.changes):
```

Comment:
Why converting the dict to a bool? Does it changes something to simply doing this?
```python
if self.ctx.changes:
```